compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -111,6 +114,18 @@ private JBallerinaBackend(PackageCompilation packageCompilation, JdkVersion jdkV
 
         // Trigger code generation
         performCodeGen();
+
+        // TODO: Move to a compiler extension once Compiler revamp is complete
+        BLangDiagnosticLog dLog = BLangDiagnosticLog.getInstance(compilerContext);
```

Comment:
Shouldn't this be run before the desugger ? If not we can move this code to executable jar write I guess.

Reply:
Moved the logic to executable Jar write in 1f3d123941e5a340e8ac91f9561e08e4316d399c